src/Handlers/Reply/ReplyHandler.ts

```diff
@@ -0,0 +1,71 @@
+import crypto from "crypto";
+import HttpMethods from "../../Constants/HttpMethods";
+import {ICredentials} from "../../Utils/Types";
+import {Hmac} from "../../Request/Hmac";
+import buckarooClient from "../../BuckarooClient";
+
+export class ReplyHandler {
+    private readonly data: object
+    private readonly uri?: string
+    private readonly auth_header?: string
+    private credentials: ICredentials;
+    private _isValid: boolean = false
+
+    constructor(credentials: ICredentials, data: string | object,auth_header?: string, uri?: string) {
+        if(typeof data === 'string'){
+            try {
+                this.data = JSON.parse(data)
+            } catch (e){
+                let objData = {}
+                new URLSearchParams(data).forEach((value, name)=>{
+                    objData[name] = value
+                })
+                this.data = objData
+            }
+        }else {
+            this.data = data
+        }
+        this.credentials = credentials
+        this.uri = uri
+        this.auth_header = auth_header
+    }
+    get isValid(){
+        return this._isValid
+    }
+    validate() {
+        if(this.data["Key"] && this.auth_header && this.uri){
+            this._isValid = this.validateJson(this.auth_header)
+        }else if (this.data["brq_signature"]){
+            this._isValid = this.validateHttp({...this.data})
+        }else {
+            throw new Error('Invalid reply data')
+        }
+        return this
+    }
+    private validateJson(auth_header:string){
+        let header = auth_header.split(':')
+        let providedHash = header[1]
+
+        let nonce = header[2]
+        let time = header[3]
+        let hmac = new Hmac(HttpMethods.POST,this.uri,this.data,nonce,time)
+
+        let hash = hmac.hashData(hmac.getHashString())
+
+        return crypto.timingSafeEqual(Buffer.from(hash),Buffer.from(providedHash))
+    }
+    private validateHttp(data:object){
+        let brq_signature = data['brq_signature']
+        let stringData = ''
+        delete data['brq_signature']
```

Comment:
"brq_signature" is not always lowercase, in plaza you are able to change the settings to upper, lower and mixed cases.